appveyor_after_build.ps1

```diff
@@ -0,0 +1,300 @@
+
+    $response = InvokeUpload "/api/profiles/restore" "backup" $qualityProfile
+}
+
+
+function SetTestQualityProfileAsDefault
+{
+    $response = InvokePostRestMethod "/api/profiles/set_as_default?language=cs&name=$testQualityProfileName" $true    
+}
+
+function VerifyAnalysisResults
+{
+    $expectedRuleViolations = @("fxcop:DoNotPassLiteralsAsLocalizedParameters", "fxcop:DoNotRaiseReservedExceptionTypes", "csharpsquid:S2228", "csharpsquid:S1134")
+
+    $response = InvokeGetRestMethod "/api/issues/search?hideRules=true"
+
+    Assert ($response.paging.total -eq 4) "There should only be 4 issues"
+    Assert ($response.issues.Length -eq 4) "There should only be 4 issues"
+
+    $brokenRules = $response.issues.rule
+    $ruleDiff = Compare-Object $response.issues.rule $expectedRuleViolations
+
+    Assert ($ruleDiff.Length -eq 0) "Expected 4 rules to be broken. Rules found: $brokenRules. Expected: $expectedRuleViolations"
+    
+    foreach ($issue in $issues)
+    {
+        Assert ($issue.project -eq $sqProjectKey) "An issue was found not belonging to the project under test. Instead, it belongs to " + $issue.project        
+    }
+}
+
+function BuildAndAnalyzeProjectUnderTest
+{
+    echo "Step 5.1: Finding msbuild"
+    $msbuildPath = FindLatestMsBuildPath
+    echo "Using the latest msbuild found from $msbuildPath"
+
+    echo "Step 5.2: Locating the MSBuild.SonarQube.Runner.exe"
+    $bootstrapperPath = FindSingleFile ([System.IO.Path]::Combine($env:APPVEYOR_BUILD_FOLDER, "DeploymentArtifacts", "BuildAgentPayload", $env:configuration)) "MSBuild.SonarQube.Runner.exe"
+
+    echo "Step 5.3: Locating the project under test"
+    $projectUnderTestPath = FindSingleFile ([System.IO.Path]::Combine($env:APPVEYOR_BUILD_FOLDER, "IntegrationTestProject")) "ProjectUnderTest.sln"
+
+    echo "Step 5.4: Waiting for SQ to have started"
+    WaitForSonarQubeToHaveStarted
+        
+    echo "Step 5.5: Upload the test quality profile and make it default"
+    UploadTestQualityProfile
+    SetTestQualityProfileAsDefault
+
+    echo "Step 5.6: Begining analysis"
+    RunProcessAndWaitForFinish $bootstrapperPath @("begin", "/k:$sqProjectKey", "/n:$sqProjectKey", "/v:1")
+
+    echo "Step 5.7: Building the project under test"
+    RunProcessAndWaitForFinish $msbuildPath @($projectUnderTestPath, "/v:n")
+
+    echo "Step 5.8: Ending analysis"
+    RunProcessAndWaitForFinish $bootstrapperPath "end"
+
+}
+
+echo "Step 1: Publishing the CS plugin as a build artifact"
+$csPluginPath = PublishPatchedCsharpPlugin
+
+echo "Step 2: Installing SonarQube"
+$sqInstallPath = FetchAndUnzipSonarQube
+
+echo "Step 3: Installing the CS plugin"
+InstallCsPlugin $sqInstallPath $csPluginPath
+
+echo "Step 4: Starting the SonarQube server"
+StartSonarQubeServer $sqInstallPath
+
+echo "Step 5: Building and analyzing the project under test"
+BuildAndAnalyzeProjectUnderTest
+
+echo "Step 6: Verify analysis results"
+VerifyAnalysisResults
+
```

Comment:
A run takes ~7 minutes, although I hope more aggressive caching will take it to ~6 minutes. We expect to run on 3 configs x 2 SQ versions, i.e. ~42 minutes. The biggest culprits are: 

Building the C# plugin via Maven  -- 1 min
Waiting for the SQ server to start -- 3 min 
The "end analysis" step -- 1 min

One possible optimization is to run the test for multiple SQ versions at once, to avoid having to wait for a new VM to become available and to also avoid building the maven plugin each time. Still, this would only shave 1 to 1.5 min for each run.